tests/phpunit/integration/Modules/Reader_Revenue_Manager/Web_TagTest.php

```diff
@@ -21,9 +21,26 @@
 class Web_TagTest extends TestCase {
 	const PUBLICATION_ID = '12345';
 
-	public function test_register() {
+	public function set_up() {
+		parent::set_up();
+
 		$web_tag = new Web_Tag( self::PUBLICATION_ID, Reader_Revenue_Manager::MODULE_SLUG );
 		$web_tag->register();
+	}
+
+	public function test_snippet_not_inserted_on_non_singular_posts() {
+		do_action( 'wp_enqueue_scripts' );
+
+		$footer_html = $this->capture_action( 'wp_footer' );
+
+		$this->assertStringNotContainsString( 'Google Reader Revenue Manager snippet added by Site Kit', $footer_html );
+		$this->assertStringNotContainsString( '<script type="text/javascript" src="https://news.google.com/swg/js/v1/swg-basic.js" id="google_swgjs-js" async="async" data-wp-strategy="async"></script>', $footer_html ); // phpcs:ignore WordPress.WP.EnqueuedResources.NonEnqueuedScript
+		$this->assertStringNotContainsString( '(self.SWG_BASIC=self.SWG_BASIC||[]).push(basicSubscriptions=>{basicSubscriptions.init({"type":"NewsArticle","isPartOfType":["Product"],"isPartOfProductId":"' . self::PUBLICATION_ID . ':openaccess","clientOptions":{"theme":"light","lang":"en-US"}});});', $footer_html );
```

Comment:
Hey @nfmohit, although I'm generally an advocate of DAMP over DRY in a testing context, in this case I feel that repeating these complex string literals makes the tests a bit harder to compare and maintain. I think we should extract them to a shared constant. In order to save a round trip I'll apply this change myself.